src/shared/bus-unit-util.c

```diff
@@ -639,43 +638,37 @@ static int bus_append_exec_command(sd_bus_message *m, const char *field, const c
                 }
         } while (!done);
 
-        if (!is_ex_prop && (flags & (EXEC_COMMAND_NO_ENV_EXPAND|EXEC_COMMAND_FULLY_PRIVILEGED|EXEC_COMMAND_NO_SETUID|EXEC_COMMAND_VIA_SHELL))) {
-                /* Upgrade the ExecXYZ= property to ExecXYZEx= for convenience */
-                is_ex_prop = true;
-
-                upgraded_name = strjoin(field, "Ex");
-                if (!upgraded_name)
-                        return log_oom();
-                field = upgraded_name;
-        }
-
-        if (is_ex_prop) {
-                r = exec_command_flags_to_strv(flags, &ex_opts);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to convert ExecCommandFlags to strv: %m");
+        if (flags & (EXEC_COMMAND_NO_ENV_EXPAND|EXEC_COMMAND_FULLY_PRIVILEGED|EXEC_COMMAND_NO_SETUID|EXEC_COMMAND_VIA_SHELL)) {
+                /* We need to use ExecXYZEx=. */
+                if (!endswith(field, "Ex"))
+                        field = strjoina(field, "Ex");
+                ex_prop = true;
+                assert_se(exec_command_flags_to_strv(flags, &ex_opts) >= 0);
+        } else {
+                if (endswith(field, "Ex"))
+                        field = strndupa(field, strlen(field) - 2);
```

Comment:
strndupa_safe

Reply:
This function is only called if the field is one of the constants listed in the table. So we know that the string is always of fixed size.

Reply:
We carry a coccinelle lint for this though. It's better to uniformly use the safe versions which are more manageable and less error-prone